JuliaExperimental/julia/gaprat.jl

```diff
@@ -10,10 +10,18 @@ module GAPRatModule
 import Base: zero, -, one, inv, ==, isless, +, *, //, ^, mod, iszero, string,
              numerator, denominator, abs, gcd
 
-import Main.GAP: GapObj
-
 export GAPRat, get_gaprat_ptr
 
+"""
+    GapObj
+
+> Holds a pointer to an object in the GAP CAS, and additionally some internal information for
+> GAP's garbage collection. It can be used as arguments for GapFunc's.
+"""
+mutable struct GapObj
+    ptr::Ptr{Cvoid}
```

Comment:
BTW, note that this is unsafe, because this void pointer is not tracked by the garbage collector; so if the only reference to some GAP object ends up on a `GapObj` struct, that object will be garbage collected, and the reference becomes stale. To fix this, all uses of `GapObj` will need to be replaced with `GAPInputType_noint` and/or `GAPInputType` (depending on context).